source/Detector/Amptek/SGM/AmptekCommandManagerSGM.h

```diff
@@ -0,0 +1,102 @@
+#ifndef AMPTEKCOMMANDMANAGERSGM_H
+#define AMPTEKCOMMANDMANAGERSGM_H
+
+#include <QObject>
+#include <QStringList>
+
+#include "Detector/Amptek/AmptekCommandManager.h"
+
+class AmptekCommandManagerSGM : public AmptekCommandManager
+{
+	Q_OBJECT
+
+public:
+	enum AmptekCommandDefSGM {
+		// acknowledge type
+		AcknowledgeOk = 0,
+		AcknowledgeOKWithInterfaceSharingRequest,
+		AcknowledgeSyncError,
+		AcknowledgePIDError,
+		AcknowledgeLENError,
+		AcknowledgeChecksumError,
+		AcknowledgeBadParameter,
+		AcknowledgeBadHexRecrod,
+		AcknowledgeUnrecognizedCommand,
+		AcknowledgeFPGAError,
+		AcknowledgeCP2201NotFound,
+		AcknowledgeScopeDataNotAvailable,
+		AcknowledgePC5NotPresent,
+		AcknowledgeI2CError,
+
+		// response type
+		ResponseRequestStatusPacket,
+		Response256ChannelSpectrumPlusStatus,
+		Response512ChannelSpectrumPlusStatus,
+		Response1024ChannelSpectrumPlusStatus,
+		ResponseConfiguration,
+		ResponseCommTestEchoPacket,
+
+		// request type
+		ReqeustStatusPacket,
+		RequestClearSpecturm,
+		RequestSpecturmPlusStatus,
+		RequestAndClearSpecturmPlusStatus,
+		RequestEnableMCAMCS,
+		RequestDisableMCAMCS,
+		RequestTextConfigurationReadback,
+		RequestTextConfiguration,
+		RequestCommTestACKOKPacket,
+		RequestCommTestACKOKWithSharingPacket,
+		RequestCommTestACKSyncErrorPacket,
+		RequestCommTestACKPIDErrorPacket,
+		RequestCommTestACKLENErrorPacket,
+		RequestCommTestACKChecksumErrorPacket,
+		RequestCommTestACKBadParameterPacket,
+		RequestCommTestEchoPacket
+	};
+
+	/// the static function to get the object
+	static AmptekCommandManagerSGM* sddCommands();
+	/// the static function to release the object
+	static void releaseSDDCommands();
+
+	virtual ~AmptekCommandManagerSGM();
+
+//	QString requestStatusPacketHex() const;
```

Comment:
These commented codes/functions will be deleted before merge.

I created a function ```requestDataPacket()``` to replace all those functions. I keep those replaced codes here for the convenience of reviewers to understand the functionality of ```requestDataPacket()```.